File-based Workflow Editor

jmchilton opened the pull request:
- Allow workflows to be uploaded via path using the GUI and and API for admins.
- Track the path of such uploads and re-sync workflows on save (both .ga and format 2 workflows) - this will allow Planemo to leverage the Galaxy workflow editor to interactively save workflows (e.g. ``planemo workflow_edit <workflow.(ga|yml)>``).

Builds on #6850

bgruening commented on test/base/populators.py:
```diff
@@ -625,6 +633,21 @@ def download_workflow(self, workflow_id, style=None):
+            workflow=workflow_object
+        )
+        raw_url = 'workflows/%s' % workflow_id
+        # url = self.galaxy_interactor._api_url(raw_url, use_key=True)
```
This line should be removed?

jmchilton committed:
Path-aware workflows.

- Allow workflows to be uploaded via path using the GUI and and API for admins.
- Track the path and resync workflows on save (both .ga and format 2 workflows) - this will allow Planemo to leverage the Galaxy workflow editor to interactively save workflows.

martenson commented:
Seems to work as expected in my testing, thanks @jmchilton
